tensorflow/core/kernels/sparse_tensor_dense_matmul_op.cc

```diff
@@ -262,53 +298,68 @@ struct SparseTensorDenseMatMulFunctor<CPUDevice, T, Tindices, ADJ_A, ADJ_B> {
 
     if (rhs_right < kNumVectorize) {
       // Disable vectorization if the RHS of output is too small
-      auto maybe_adjoint_b = MaybeAdjoint<decltype(b), ADJ_B>(b);
+      auto maybe_adjoint_b = MaybeAdjoint<decltype(b), ADJ_B, NDIM>(b);
 
       for (std::size_t i = 0; i < nnz; ++i) {
-        const Tindices m = internal::SubtleMustCopy(a_indices(i, lhs_index_a));
-        const Tindices k = internal::SubtleMustCopy(a_indices(i, rhs_index_a));
+        Eigen::array<Eigen::Index, NDIM> indices;
+        for (int j = 0; j < NDIM; ++j) {
+          indices[j] = internal::SubtleMustCopy(a_indices(i, j));
+        }
```

Comment:
Actually, we should do a bounds check on all the indices instead of just m and k.